### PR TITLE
Add self-close linting rule for components with no children

### PR DESCRIPTION
**Pull request by aekong**
enforces rule for self-closing components with no children

### CODE REVIEW

**Comment by aekong** [comment]
@james-wills-sf is this the correct linting rule you mentioned in https://github.com/SFDigitalServices/sf-dahlia-lap/pull/432#discussion_r512226902?

> components with no children should use the closing shorthand like `<div className='some-div' />`

cc @akegan

File: .eslintrc.js [hunk 1]
```diff
@@ -75,6 +75,10 @@ module.exports = {
     'react/prop-types': 'off',
     'react/display-name': 'off',
     'react/state-in-constructor': 0,
+    'react/self-closing-comp': ['error', {
+      'component': true,
```
Comment (james-wills-sf) [inline .eslintrc.js:79]: since component/html=true is the default, you can omit them and just make the above line: `"react/self-closing-comp": "error"`

**Review by james-wills-sf — COMMENTED** [review]
@aekong thanks for adding this rule! 

2 questions:
1. why are there 9 commits on this PR, shouldn't there just be one? Will that change once your other PR is merged?
2. Did you run yarn lint:fix? I'd be surprised if this rule isn't broken anywhere in our codebase

**Comment by akegan** [comment]
If there are linting errors that need to be fixed, shouldn't circleCI be failing?

**Comment by james-wills-sf** [comment]
Also just making sure: do you have the pre-commit hook working where it doesn't let you commit if there are any linting errors?

Edit: if you don't just run `overcommit --install`

Again I'm just suspicious that no code in our codebase needed to be fixed after adding this rule

and @akegan , I actually can't remember if we added linting as a CI check, or did we just do it as a pre-commit hook?

File: .eslintrc.js (hunk 1 above)
Comment (aekong) [reply .eslintrc.js:79]: good to know!

**Comment by aekong** [comment]
weird! i already merged that `atom-tests` branch.  i wonder if it's because i branched off my `atom-tests` branch to test the rule out.  i'll make a new branch off `main` and bring these changes in.

yeah, i have precommit set up (think it got set up for free when i set up the project initially).

i'll revisit this later today.  thanks for looking, all 👍

**Comment by akegan** [comment]
@james-wills-sf yep I just double checked and it's in there. We had issues in the past with only having it as a pre-commit hook because it's easy to bypass it plus the pre-commit hook should in theory only check the changed files to speed it up

**Comment by aekong** [comment]
closing in favor of #446

**Pull request closed without merging**